src/cast_core/src/Loader.cpp

```diff
@@ -38,27 +38,68 @@ using index_type = std::pair<DocumentGenerator, std::optional<DocumentGenerator>
 
 /** @private */
 struct Loader::PhaseConfig {
-    PhaseConfig(PhaseContext& context, mongocxx::pool::entry& client, uint thread, ActorId id)
+    PhaseConfig(PhaseContext& context,
+                mongocxx::pool::entry& client,
+                uint thread,
+                size_t totalThreads,
+                ActorId id)
         : database{(*client)[context["Database"].to<std::string>()]},
-          // The next line uses integer division. The Remainder is accounted for below.
-          numCollections{context["CollectionCount"].to<IntegerSpec>() /
-                         context["Threads"].to<IntegerSpec>()},
+          multipleThreadsPerCollection{
+              context["MultipleThreadsPerCollection"].maybe<bool>().value_or(false)},
+          // The next line uses integer division for non multithreaded configurations.
+          // The Remainder is accounted for below.
+          numCollections{multipleThreadsPerCollection
```

Comment:
Ack. Could you add a comment saying we are using an explicit option to avoid breaking existing functionality?

Reply:
will do, I think that we should also open another new ticket to try to merge the functionality as you are suggesting. But I suspect that it may not be straight forward (mainly because there are existing tests).

Reply:
Only open the ticket if there seems to be worth pushing on at some point. Otherwise, just document and move on.